daemon/src/daemon.rs

```diff
@@ -509,76 +531,93 @@ impl DocumentActor {
 
-    async fn process_cursor_states(&mut self, cursor_states: Vec<CursorState>) {
+    async fn send_cursor_states_to_editors(
+        &mut self,
+        exclude_id: Option<EditorId>,
+        cursor_states: Vec<CursorState>,
+    ) {
         for CursorState {
-            userid,
+            cursor_id,
             name,
             file_path,
             ranges,
         } in cursor_states
         {
-            let message = EditorProtocolMessageToEditor::Cursor {
-                userid,
+            let message = EditorProtocolObject::Request(EditorProtocolMessageToEditor::Cursor {
+                cursor_id,
                 name,
                 uri: format!("file://{}", self.absolute_path_for_file_path(&file_path)),
                 ranges,
-            };
-            self.send_to_editor_clients(message).await;
+            });
+            let editor_ids: Vec<EditorId> = self.ot_servers.keys().cloned().collect();
+            for editor_id in editor_ids {
+                if let Some(exclude_id) = exclude_id {
+                    if editor_id == exclude_id {
+                        continue;
+                    }
+                }
+
+                self.send_to_editor_client(&editor_id, message.clone())
+                    .await;
+            }
         }
     }
 
-    async fn send_to_editors(&mut self, rev_delta: RevisionedEditorTextDelta, file_path: &str) {
+    async fn send_to_editor(
+        &mut self,
+        editor_id: &EditorId,
+        rev_delta: RevisionedEditorTextDelta,
+        file_path: &str,
+    ) {
         let message = EditorProtocolMessageToEditor::Edit {
             uri: format!("file://{}", self.absolute_path_for_file_path(file_path)),
             delta: rev_delta,
         };
-        self.send_to_editor_clients(message).await;
+        self.send_to_editor_client(editor_id, EditorProtocolObject::Request(message))
+            .await;
     }
 
     async fn send_to_editor_client(&mut self, editor_id: &EditorId, message: EditorProtocolObject) {
         if let Some(handle) = self.editor_clients.get_mut(editor_id) {
             if handle.send(message).await.is_err() {
-                info!("Removing EditorHandle from client list.");
-                self.editor_clients.remove(editor_id);
-            }
-        } else {
-            warn!(
-                "Sending to editor client failed: We don't have a client registered with id #{}.",
-                editor_id.0
-            );
-        }
-    }
-
-    async fn send_to_editor_clients(&mut self, message: EditorProtocolMessageToEditor) {
-        let mut to_remove = Vec::new();
-        for (id, handle) in &mut self.editor_clients.iter_mut() {
-            if handle
-                .send(EditorProtocolObject::Request(message.clone()))
-                .await
-                .is_err()
-            {
+                warn!(
+                    "Sending to editor client failed: We don't have a client registered with id #{editor_id:?}"
```

Comment:
Tried addressing that in ed09de7, does that make more sense? The warn! message was indeed not fitting. I moved the original one to the place it belongs, and made this one better (and lowered it to an info!, because it can happen during regular editor disconnect).